app/models/miq_report/search.rb

```diff
@@ -104,6 +104,8 @@ def paged_view_search(options = {})
 
     if options[:parent]
       targets = get_parent_targets(options)
+      # filter out archived VMs
+      targets = targets.select { |target| !Vm.find_by(:id => target).archived? unless Vm.find_by(:id => target).nil? }
```

Comment:
@chessbyte as I'm looking on it now, I'm considering to use [`try`](http://apidock.com/rails/Object/try) Rails method  , so this line would become

```ruby
targets = targets.select { |target| !Vm.find_by(:id => target).try(:archived?) }
```

but that would mean, I'd need to do the same with condition below, checking if `target` variable is `empty?`.

Any better suggestions? @matthewd @Fryguy